File: Algorithms/435. Non-overlapping Intervals - Method 2.py
# Given an array of intervals intervals where intervals[i] = [starti, endi], 
# return the minimum number of intervals you need to remove to make the rest of the intervals non-overlapping.


class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key = lambda x: x[1])
        output = 0
        pre_end = intervals[0][1]
        for start, end in intervals[1:]:
            if start >= pre_end:
                pre_end = end
            else:
                output += 1
        return output

File: Algorithms/435. Non-overlapping Intervals - Method 2.java
// Given an array of intervals intervals where intervals[i] = [starti, endi], 
// return the minimum number of intervals you need to remove to make the rest of the intervals non-overlapping.


class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[1], b[1]));
        int output = 0;
        int pre_end = Integer.MIN_VALUE;

        for (int i = 0; i < intervals.length; i++) {
            if (intervals[i][0] >= pre_end) {
                pre_end = intervals[i][1];
            }
            else {
                output++;
            }
        }

        return output;
    }
}
